terracotta/drivers/raster_base.py

```diff
@@ -36,6 +38,42 @@
 logger = logging.getLogger(__name__)
 
 
+class LFUCacheWithCompression(LFUCache):
+    """ Least-frequently-used Cache with data compression
+    """
+
+    def __getitem__(self, key: Any) -> np.ma.MaskedArray:
+        compressed_item = super().__getitem__(key)
+        return self._decompress_masked_array(compressed_item)
+
+    def __setitem__(self, key: Any, value: np.ma.MaskedArray) -> None:
+        super().__setitem__(key, self._compress_masked_array(value))
+
+    def _compress_masked_array(self, array: np.ma.MaskedArray) -> Tuple:
```

Comment:
The return type here should be `Tuple[np.ndarray, np.ndarray, str, Tuple[int, int]]`

Reply:
I think it might be cleaner and more expressive to define a `CompressedArray` type (and set it to this value).

Reply:
I agree, maybe a NamedTuple.

Reply:
In one-off cases like this I usually don't bother with custom data types, but we can leave it to @panakouris :wink: